### PR TITLE
ADSB: add msgs for IDENT, Registration/Callsign string, and Flight ID

### PR DESCRIPTION
**Pull request by magicrub**
Added ADSB-out msgs/cmds:
- MAV_CMD_DO_START_ADSB_OUT_IDENT to trigger an IDENT event 
- Standalone Registration/Callsign string
- Standalone Flight ID string

### CODE REVIEW

File: message_definitions/v1.0/uAvionix.xml [hunk 1]
```diff
@@ -118,5 +118,13 @@
       <description>Transceiver heartbeat with health report (updated every 10s)</description>
       <field type="uint8_t" name="rfHealth" enum="UAVIONIX_ADSB_RF_HEALTH" display="bitmask">ADS-B transponder messages</field>
     </message>
+    <message id="10004" name="UAVIONIX_ADSB_OUT_CFG_REGISTRATION">
+      <description>Aircraft Registration. This is often referred to as the callsign for ADSB-Out vehicles.</description>
+      <field type="char[9]" name="registration">Aircraft Registration (ASCII string A-Z, 0-9 only), e.g. “N8644B “. Trailing spaces (0x20) only. This is null-terminated.</field>
```
Comment (meee1) [inline message_definitions/v1.0/uAvionix.xml:123]: can i request the removal of utf8 characters?

Comment (magicrub) [reply message_definitions/v1.0/uAvionix.xml]: yup, those snuck in from a pdf copy/paste. Just removed them. thanks!

File: message_definitions/v1.0/uAvionix.xml [hunk 2]
```diff
@@ -82,6 +82,17 @@
...
+      <entry value="54132" name="MAV_CMD_DO_START_ADSB_OUT_IDENT" hasLocation="false" isDestination="false">
+        <description>Trigger the start of an ADSB-out IDENT. This should only be used when requested to do so by an Air Traffic Controller in controlled airspace. This starts the IDENT which is then typically held for 18 seconds by the ADSB-out hardware per the ADSB spec.</description>
+        <param index="1">Empty</param>
+        <param index="2">Empty</param>
+        <param index="3">Empty</param>
+        <param index="4">Empty</param>
+        <param index="5">Empty</param>
+        <param index="6">Empty</param>
+        <param index="7">Empty</param>
+      </entry>
```
Comment (meee1) [inline message_definitions/v1.0/uAvionix.xml:95]: malformed xml

Comment (magicrub) [reply message_definitions/v1.0/uAvionix.xml:95]: whoops. I fixed that but forgot to push. All fixed now!

**Comment by magicrub** [comment]
The DO_IDENT msg is already in Mission Planner beta
image

File: message_definitions/v1.0/uAvionix.xml [hunk 3]
```diff
@@ -118,5 +130,17 @@
       <description>Transceiver heartbeat with health report (updated every 10s)</description>
       <field type="uint8_t" name="rfHealth" enum="UAVIONIX_ADSB_RF_HEALTH" display="bitmask">ADS-B transponder messages</field>
     </message>
+    <message id="10004" name="UAVIONIX_ADSB_OUT_CFG_REGISTRATION">
+      <description>Aircraft Registration.</description>
+      <field type="char[9]" name="registration">Aircraft Registration (ASCII string A-Z, 0-9 only), e.g. "N8644B ". Trailing spaces (0x20) only. This is null-terminated.</field>
```
Comment (peterbarker) [inline message_definitions/v1.0/uAvionix.xml:135]: We're using "callsign" in `UAVIONIX_ADSB_OUT_CFG` for this, I think?  We should be consistent in our naming of the same data.  And our descriptions for that matter, given the trailing spaces comment here.  This also affects this message's name.

```suggestion
      <field type="char[9]" name="registration">Aircraft Registration (ASCII string A-Z, 0-9 only), e.g. "N8644B ". Spaces (0x20) can only appear at end of string. This is null-terminated.</field>
```

Similarly for the other 2 messages here.

File: message_definitions/v1.0/uAvionix.xml [hunk 4]
```diff
@@ -118,5 +130,17 @@
       <description>Transceiver heartbeat with health report (updated every 10s)</description>
       <field type="uint8_t" name="rfHealth" enum="UAVIONIX_ADSB_RF_HEALTH" display="bitmask">ADS-B transponder messages</field>
     </message>
+    <message id="10004" name="UAVIONIX_ADSB_OUT_CFG_REGISTRATION">
+      <description>Aircraft Registration.</description>
```
Comment (peterbarker) [inline message_definitions/v1.0/uAvionix.xml:134]: When is this message sent or received?  Note the other messages in here tell you a bit about that.

File: message_definitions/v1.0/uAvionix.xml [hunk 5]
```diff
@@ -118,5 +130,17 @@
...
+      <field type="char[9]" name="registration">Aircraft Registration (ASCII string A-Z, 0-9 only), e.g. "N8644B ". Trailing spaces (0x20) only. This is null-terminated.</field>
+    </message>
+    <message id="10005" name="UAVIONIX_ADSB_OUT_CFG_FLIGHTID">
+      <description>Flight Identification for ADSB-Out vehicles.</description>
+      <field type="char[9]" name="flight_id">Flight Identification: 8 ASCII characters, '0' through '9', 'A' through 'Z' or space. Spaces (0x20) used as a trailing pad character, or when call sign is unavailable. Reflects Control message setting. This is null-terminated.</field>
+    </message>
+    <message id="10006" name="UAVIONIX_ADSB_GET">
+      <description>Request messages.</description>
+      <field type="uint32_t" name="ReqMessageId">Message ID to request. Supports any message in this 10000-10099 range</field>
+    </message>
```
Comment (peterbarker) [inline message_definitions/v1.0/uAvionix.xml:144]: Just use `MAV_CMD_REQUEST_MESSAGE`

File: message_definitions/v1.0/uAvionix.xml [hunk 6]
```diff
@@ -82,6 +82,18 @@
       <entry value="6" name="UAVIONIX_ADSB_OUT_DOWNED_AIRCRAFT_EMERGENCY"/>
       <entry value="7" name="UAVIONIX_ADSB_OUT_RESERVED"/>
     </enum>
+    <enum name="MAV_CMD">
+      <entry value="54132" name="MAV_CMD_DO_START_ADSB_OUT_IDENT" hasLocation="false" isDestination="false">
```
Comment (peterbarker) [inline message_definitions/v1.0/uAvionix.xml:86]: That's a strange number to use.

Perhaps this `MAV_CMD` range could start at "10000" in the same way the uavionix messages do?  @hamishwillee thoughts?

File: message_definitions/v1.0/uAvionix.xml [hunk 7]
```diff
@@ -82,6 +82,18 @@
...
+        <description>Trigger the start of an ADSB-out IDENT. This should only be used when requested to do so by an Air Traffic Controller in controlled airspace. This starts the IDENT which is then typically held for 18 seconds by the ADSB-out hardware per the ADSB spec.</description>
```
Comment (peterbarker) [inline message_definitions/v1.0/uAvionix.xml:87]: This is a good description.

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: message_definitions/v1.0/uAvionix.xml (hunk 3 above)
Comment (magicrub) [reply message_definitions/v1.0/uAvionix.xml:135]: I pulled that text from their datasheet

File: message_definitions/v1.0/uAvionix.xml (hunk 6 above)
Comment (magicrub) [reply message_definitions/v1.0/uAvionix.xml:86]: yup, it was quite a keyboard smash number. I'll change it to a 10k number.

File: message_definitions/v1.0/uAvionix.xml (hunk 5 above)
Comment (magicrub) [reply message_definitions/v1.0/uAvionix.xml:144]: ok, yeah that makes sense. I've never used that before but it's something we should use more. Its kind of a pain with the whole interpretation thing.. but I can figure it out

File: message_definitions/v1.0/uAvionix.xml (hunk 4 above)
Comment (magicrub) [reply message_definitions/v1.0/uAvionix.xml:134]: at the request of uavionix, we're going to remove this. Its actually harder for them to get their stuff FAA certified if the normal pilot "tools" set everything. They want to have a different app set set-once type things which is an app they're making.

**Comment by peterbarker** [comment]
On Fri, 9 Jul 2021, Tom Pittenger wrote:

> I pulled that text from their datasheet

Right?

**Comment by hamishwillee** [comment]
Should any of this move into standard/common.xml? That already has some ADSB stuff.

**Comment by magicrub** [comment]
replaced with a simpler version https://github.com/ArduPilot/mavlink/pull/215

File: message_definitions/v1.0/uAvionix.xml (hunk 6 above)
Comment (hamishwillee) [reply message_definitions/v1.0/uAvionix.xml:86]: Sorry, missed that. No problem by me as long as new number isn't a clash.

**Pull request closed without merging**